sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -253,7 +251,16 @@ def _delay_failure(start_time: datetime.datetime) -> None:
         time.sleep((min_time - (current_time - start_time)).total_seconds())
 
 
-def _get_headers(headers, request_type, replica_count, uses_feature_flags, feature_filters_used, uses_key_vault) -> str:
+def __update_correlation_context_header(
```

Comment:
I am now starting to wonder if this method should be inside the class? 
We are not using it anywhere else and we seem to be passing a lot of class instance variables to it. 

It would reduce the number of arguments we have to pass to it.
What do you think?

Reply:
It's used in the Async version too. I plan to clean up this class eventually, splitting it maybe into a utility class and a shared class for sync/async. But need time and not another PR that would collide with it.